_posts/2022-06-23-S3-Log-Ingestion-Using-Data-Prepper-1.5.0.md

```diff
@@ -0,0 +1,138 @@
+---
+layout: post
+title:  "S3 Log Ingestion Using Data Prepper 1.5.0"
+authors:
+- dlv
+date: 2022-06-23 12:00:00 -0500
+categories:
+  - technical-post
+twittercard:
+  description: "You can start using Data Prepper 1.5.0 to ingest log data from S3 today."
+---
+
+Data Prepper is an open-source data collector for data ingestion into OpenSearch. It currently supports trace analytics 
+and log analysis use-cases. Earlier this year Data Prepper added log ingestion over HTTP using tools such as 
+[Fluent Bit](https://fluentbit.io/).
+
+Today, the Data Prepper maintainers announce the release of Data Prepper 1.5.0 with support of 
+[Amazon S3](https://aws.amazon.com/s3/) as a source.
+
+
+## The Current Environment
+
+Many teams use cloud object stores such as Amazon S3 for storing logs. Many AWS services write logs to Amazon S3 that 
+customers want to analyze. For example, AWS Application Load Balancer writes access logs to S3. As part of a 
+comprehensive log solution, teams want to incorporate this log data along with their application logs.
+
+It’s not only AWS services writing logs to S3. S3 is a highly available service offering that does a fantastic job of 
+taking in large volumes of data. Because it does this so well, some application developers are sending their logs to S3.
+
+Right now, getting this log data out of S3 is complicated. Developers are writing their own code to read from S3. Much 
+of this is duplicated code for receiving S3 Event Notifications and then parsing S3. And developers may also easily hit 
+issues with the size and scale of some files.
+
+
+## Go Where the Logs Are
+
+To solve this recurring issues for teams, Data Prepper 1.5 adds support for Amazon S3 as a source of log data. S3 has a 
+feature for S3 Event Notifications that Data Prepper leverages to get log data. S3 Event Notifications send 
+notifications to configured destinations whenever objects in an S3 bucket change. For example, if a new object is 
+created, S3 will send this notification. You can configure an S3 bucket to send event notifications to an 
+[Amazon SQS](https://aws.amazon.com/sqs/) queue whenever new objects are written to S3. You then configure Data Prepper 
+to use that SQS queue for receiving event notifications.
+
+Data Prepper will receive messages from the SQS queue. For any newly created object, it then gets that object out of S3 
+and parses it into events. Initially, Data Prepper can read two types of formats.
+
+1. Single-line logs. These are logs where a single log line indicates the same event.
+2. JSON objects. Data Prepper expects a common JSON pattern where the JSON has one large JSON array of smaller objects. Data Prepper will take each smaller object and create a single event from that object.
+
+Additionally, Data Prepper supports either uncompressed data or Gzip-compressed data.
+
+
+### An Example
+
+Earlier, I mentioned supporting Application Load Balancer (ALB) access logs. These logs are saved as traditional logs. 
```

Comment:
I use "ALB" in the next sentence. Should I remove it still?

Reply:
We should almost always spell out Application Load Balancer. I know it may seem wordy, but it's the proper name of the resource and is what we use in the AWS documentation. On the consoles, when space is limited (such as dropdown lists), it’s acceptable to use the ALB acronym if the context is clear.